kernel/kernel-api/src/main/java/io/delta/kernel/client/JsonHandler.java

```diff
@@ -98,4 +99,24 @@ CloseableIterator<ColumnarBatch> readJsonFiles(
         CloseableIterator<FileStatus> fileIter,
         StructType physicalSchema,
         Optional<Predicate> predicate) throws IOException;
+
+    /**
+     * Write each `Row` in given `data` serialized as JSON and written as separate line in
+     * destination file. This call either succeeds in creating the file with given contents or no
+     * file is created at all.
+     * <p>
+     * Following are the supported data types:
+     * {@code boolean}, {@code byte}, {@code short}, {@code int}, {@code long}, {@code float},
+     * {@code double}, {@code string}, {@code StructType} (containing any of the supported types as
+     * subtypes), {@code ArrayType}, {@code MapType} (only a map with string keys is supported).
+     * <p>
+     *
+     * @param filePath Fully qualified destination file path
+     * @param data     Iterator of {@link Row} objects where each row should be serialized as JSON
+     *                 and written as separate line in the destination file.
+     *                 <p>
+     * @throws FileAlreadyExistsException if the file already exists.
+     * @throws IOException if any other I/O error occurs.
+     */
+    void writeJsonFileAtomically(String filePath, CloseableIterator<Row> data) throws IOException;
```

Comment:
Wondering if we need to be more explicit about the expected behavior here. Is the Jackson behavior for nulls in map/arrays standard JSON serialization rules or should we be specific here? The different JSON serialization protocols is confusing and I'm not sure how we can make it less confusing for other developers/connectors.

Reply:
Can you maybe create a github issue for this? We should clarify the two different cases in either the protocol or user guide.